pkg/master/master.go

```diff
@@ -155,6 +180,58 @@ type completedConfig struct {
 	*Config
 }
 
+func (c *Config) createMasterCountReconciler() EndpointReconciler {
+	// use a default endpoint reconciler if nothing is set
+	endpointClient := coreclient.NewForConfigOrDie(c.GenericConfig.LoopbackClientConfig)
+	return NewMasterCountEndpointReconciler(c.MasterCount, endpointClient)
+}
+
+func (c *Config) createNoneReconciler() EndpointReconciler {
+	return NewNoneEndpointReconciler()
+}
+
+func (c *Config) createLeaseReconciler() EndpointReconciler {
+	ttl := c.MasterEndpointReconcileTTL
+	config, err := c.StorageFactory.NewConfig(kapi.Resource("apiServerIPInfo"))
+	if err != nil {
+		glog.Fatalf("Error determining service IP ranges: %v", err)
+	}
+	leaseStorage, _, err := storagefactory.Create(*config)
+	if err != nil {
+		glog.Fatalf("Error creating storage factory: %v", err)
+	}
+	endpointConfig, err := c.StorageFactory.NewConfig(kapi.Resource("endpoints"))
+	if err != nil {
+		glog.Fatalf("Error getting storage config: %v", err)
+	}
+	endpointsStorage := endpointsstorage.NewREST(generic.RESTOptions{
+		StorageConfig:           endpointConfig,
+		Decorator:               generic.UndecoratedStorage,
+		DeleteCollectionWorkers: 0,
+		ResourcePrefix:          c.StorageFactory.ResourcePrefix(kapi.Resource("endpoints")),
```

Comment:
I would use something that clearly won't collide with the endpoints api. `"kube-apiserver-endpoint"`?

Reply:
@ncdc I don't remember whether there was a reason we didn't do that before?